### ISSUE TITLE
Unable to commit when git hooks use bash syntax.

### ISSUE THREAD

**Issue by b-hayes** [desktop/desktop#10060]
### Describe the feature or problem you’d like to solve

Our project have git hooks that were written for Mac and Linux but I am in windows. WSL is my dev env and can run these hooks but Github desktop will not run any git commands in WSL behind the scenes.

Github desktop is unusable by many devs using WSL.

When a git hook is run via GitHub desktop we see:
Commit failed - exit code 1 received, with output: ''.' is not recognized as an internal or external command,
operable program or batch file.'

Also tried using bash as my shell.
Also tried adding "bash -c" etc into the git hooks that run but then I get "bash is not a recognized command" even tho it IS recognized by all my shells.

### Proposed solution

Please add WSL integration || or actually use the shell to run already working git cli so that desktop is just an interface running git commands behind the scenes would be nice.

### Additional context

Some context.. the hooks are php files with 
#!/usr/bin/env php
<?php

They run additional commands also with shell executions/passthrough.
I tried editing those to add `bash -c` etc however and see "bash is not a recognized command"
Even who it IS a recognized command in windows and works for all my shells. 🤷‍♀️

I have seen people talking about editing the source to add custom shells but its easier to just not use Github.

I don't have the option of not using WSL or editing the git hooks to support windows at this stage.

**Comment by tierninho** [comment]
@b-hayes Thanks for the issue. We added support for opening a WSL shell from within GitHub Desktop in #7821.

Are you running into these issues with mac or Linux? We only support Mac on this channel. A former developer of GitHub Desktop maintains a fork with Linux builds available for download:

https://github.com/shiftkey/desktop/releases

We are also tracking official Linux support in #1525.

**Comment by b-hayes** [comment]
I'm on windows, hence WSL question.

I didn't WSL option but from what you just said I think I misunderstood the shell integration options entirely.

I thought it was using the shell environment to run git commands. Eg. my git hooks would work if WSL was an option.

I am aware there is a Linux version and I could possibly run this via xserver from WSL itself as a convoluted workaround but then git hooks don't work on GitHub for Desktop in Linux either lol. (last time I checked had to tell Ubuntu users to abandon Github desktop and use something else).

So im just out of luck it seems.

But I do think this could fix a lot of things across all systems if the git commands were run in the user's shell of choice.

**Comment by tierninho** [comment]
Oops, I totally blanked on the recognizing it was a Windows env, as I was thrown off with the Mac and Linux comment. 😝 

> But I do think this could fix a lot of things across all systems if the git commands were run in the user's shell of choice.

I do think you are in luck as I just spotted this: https://github.com/desktop/desktop/pull/8863 which was just released today!

**Comment by b-hayes** [comment]
> Oops, I totally blanked on the recognizing it was a Windows env, as I was thrown off with the Mac and Linux comment. 😝
> 
> > But I do think this could fix a lot of things across all systems if the git commands were run in the user's shell of choice.
> 
> I do think you are in luck as I just spotted this: #8863 which was just released today!

I see that issue says "Fixed not being able to run the GitHub command from the WSL CLI" which is not what I am talking about.
I meant that git commands are run in the user's shell behind the scenes to avoid all the issues.

Example:
In Github desktop, I click commit and it runs `WSL "git commit -m \"the commit message\""`

Running git itself in the users chosen shell means it bypasses having to add special support for the bash syntax or detecting and running git hooks in the appropriate manner according to the shebang lines in the script.
Doing it this way means what shell is installed or what version of bash.exe is used doesn't matter. The correct version of PHP or node or anything else needed to be run by a shebang line will be of no concern if native git command is run in the wsl ditrol.

Just bypass all of it and use Github desktop to execute commands for you as if you were in the terminal.

**Comment by b-hayes** [comment]
Came back to GitHub Desktop as I miss it and nothing has changed. Github desktop is still useless to anyone using WSL.
Git hooks do not run. Status is really slow takes ages for changes to appear in the window, and it often creates a file called 'NUL' inside your project whenever you do a commit. Then you have to keep deleting this file all the time. 🤦‍♂️

Loving WSLv2 its a massive improvement but still can't use the Github desktop app 😢

**Comment by ghost** [comment]
Any updates on this? I'm too facing this very same issue, where I can't run githooks on my windows environment if the repo and its necessities are handled in wsl (node, PHP, make, etc.)

**Comment by JohnCliffe** [comment]
added git hooks to WSL project so guess I need to move on from github desktop then?

**Comment by b-hayes** [comment]
> added git hooks to WSL project so guess I need to move on from github desktop then?

yeah pretty much. 2 years later and we still cant use github desktop with wsl at all.

**Comment by niik** [comment]
Yesterday we shipped GitHub Desktop 3.5.5-beta1 to our [beta channel](https://desktop.github.com/beta/) with [improved support for hooks related to committing](https://github.com/desktop/desktop/pull/21319).

<img width="280" alt="Screenshot showing the GitHub Desktop commit form as a pre-commit hook is running" src="https://github.com/user-attachments/assets/b389d45d-90d5-4660-9bab-f9b6b64c8201" />

The support is experimental and currently limited to our beta channel. While we believe that this release resolved this issue we're looking for real-world feedback so if you have the ability we'd appreciate it if you could download our [beta channel](https://desktop.github.com/beta/) and give it a spin. Ensure the experimental hooks support it's enabled by going to Settings -> Git -> Hooks and making sure the "Load Git hook environment variables from shell" is ticked.

<img width="614" height="558" alt="Screenshot showing the hook settings dialog" src="https://github.com/user-attachments/assets/4ed1157c-1c89-4733-88db-dd8fe84cf58f" />
